dandiapi/api/tests/factories.py

```diff
@@ -75,21 +76,38 @@ def extra_data(self):
 class DandisetFactory(factory.django.DjangoModelFactory):
     class Meta:
         model = Dandiset
+        skip_postgeneration_save = True
+
+    # Can't use RelatedFactoryList for owners: https://github.com/FactoryBoy/factory_boy/issues/544
+    @factory.post_generation
+    def owners(self, create: bool, extracted: list[User] | None, **kwargs) -> None:  # noqa: FBT001
+        if not create:
+            return
+        if extracted is None:
+            extracted = []
+        for user in extracted:
+            add_dandiset_owner(self, user)
 
 
 class BaseVersionFactory(factory.django.DjangoModelFactory):
     class Meta:
+        model = Version
+        skip_postgeneration_save = True
         abstract = True
 
     dandiset = factory.SubFactory(DandisetFactory)
     name = factory.Faker('sentence')
+    # Don't use "Version.next_published_version" to create versions, as that will require
+    # additional database queries.
+    version = factory.Sequence(
```

Comment:
In tests, `Version` created at the exact same time via `VersionFactory` will no longer have the same `.version` value. I think this is actually an improvement as it's more realistic. If a duplicate version is desired, it can still be created with `VersionFactory(version=...)`.